cmd/settings.go

```diff
@@ -32,17 +32,22 @@ type settings struct {
 	PrivacyAllowWipe          bool     `json:"privacy.allow_wipe"`
 	PrivacyExportable         []string `json:"privacy.exportable"`
 
-	UploadProvider             string `json:"upload.provider"`
-	UploadFilesystemUploadPath string `json:"upload.filesystem.upload_path"`
-	UploadFilesystemUploadURI  string `json:"upload.filesystem.upload_uri"`
-	UploadS3AwsAccessKeyID     string `json:"upload.s3.aws_access_key_id"`
-	UploadS3AwsDefaultRegion   string `json:"upload.s3.aws_default_region"`
-	UploadS3AwsSecretAccessKey string `json:"upload.s3.aws_secret_access_key,omitempty"`
-	UploadS3Bucket             string `json:"upload.s3.bucket"`
-	UploadS3BucketDomain       string `json:"upload.s3.bucket_domain"`
-	UploadS3BucketPath         string `json:"upload.s3.bucket_path"`
-	UploadS3BucketType         string `json:"upload.s3.bucket_type"`
-	UploadS3Expiry             string `json:"upload.s3.expiry"`
+	UploadProvider             string            `json:"upload.provider"`
+	UploadFilesystemUploadPath string            `json:"upload.filesystem.upload_path"`
+	UploadFilesystemUploadURI  string            `json:"upload.filesystem.upload_uri"`
+	UploadS3AwsAccessKeyID     string            `json:"upload.s3.aws_access_key_id"`
+	UploadS3AwsDefaultRegion   string            `json:"upload.s3.aws_default_region"`
+	UploadS3AwsSecretAccessKey string            `json:"upload.s3.aws_secret_access_key,omitempty"`
+	UploadS3Bucket             string            `json:"upload.s3.bucket"`
+	UploadS3BucketDomain       string            `json:"upload.s3.bucket_domain"`
+	UploadS3BucketPath         string            `json:"upload.s3.bucket_path"`
+	UploadS3BucketType         string            `json:"upload.s3.bucket_type"`
+	UploadS3Expiry             string            `json:"upload.s3.expiry"`
+	UploadDavEndpoint          string            `json:"upload.webdav.endpoint"`
+	UploadDavUsername          string            `koanf:"upload.webdav.username"`
+	UploadDavPassword          string            `koanf:"upload.webdav.password"`
+	UploadDavRootPath          string            `koanf:"upload.webdav.root_path"`
+	UploadDavHeaders           map[string]string `koanf:"upload.webdav.headers"`
 
```

Comment:
Should we add 

```if set.UploadDavEndpoint != "" 
	if !dav.Connect(dav.Opts{
		Endpoint: set.UploadDavEndpoint,
		Username: set.UploadDavUsername,
		Password: set.UploadDavPassword,
		RootPath: set.UploadDavRootPath,
		Headers:  set.UploadDavHeaders,
	}) {
		return echo.NewHTTPError(http.StatusBadRequest,
			fmt.Sprintf("Error connecting to webdav provider: %v", err))
	}
```

in line number 187 as an additional validation for configurations